bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/state/PowermaxState.java

```diff
@@ -12,210 +12,101 @@
-        if ((zone >= 1) && (zone <= zones.length)) {
-            this.zones[zone - 1].setLastTripped(lastTripped);
-        }
-    }
-
-    /**
-     * Compare the sensor last trip with a given time
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     * @param refTime the time in ms to compare with
-     *
-     * @return true if the sensor is tripped and last trip is older than the given time
-     */
-    public boolean isLastTripBeforeTime(int zone, long refTime) {
-        return ((zone < 1) || (zone > zones.length)) ? false : zones[zone - 1].isLastTripBeforeTime(refTime);
-    }
-
-    /**
-     * Get whether or not the battery of the zone sensor is low
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     *
-     * @return true when the battery is low
-     */
-    public Boolean isSensorLowBattery(int zone) {
-        return ((zone < 1) || (zone > zones.length)) ? null : zones[zone - 1].isLowBattery();
-    }
-
-    /**
-     * Set whether or not the battery of the zone sensor is low
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     * @param lowBattery true if battery is low
-     */
-    public void setSensorLowBattery(int zone, Boolean lowBattery) {
-        if ((zone >= 1) && (zone <= zones.length)) {
-            this.zones[zone - 1].setLowBattery(lowBattery);
-        }
-    }
-
-    /**
-     * Get whether or not the zone sensor is bypassed
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     *
-     * @return true if bypassed
-     */
-    public Boolean isSensorBypassed(int zone) {
-        return ((zone < 1) || (zone > zones.length)) ? null : zones[zone - 1].isBypassed();
-    }
-
-    /**
-     * Set whether or not the zone sensor is bypassed
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     * @param bypassed true if bypassed
-     */
-    public void setSensorBypassed(int zone, Boolean bypassed) {
-        if ((zone >= 1) && (zone <= zones.length)) {
-            this.zones[zone - 1].setBypassed(bypassed);
-        }
-    }
-
-    /**
-     * Get whether or not the zone sensor is armed
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     *
-     * @return true if armed
-     */
-    public Boolean isSensorArmed(int zone) {
-        return ((zone < 1) || (zone > zones.length)) ? null : zones[zone - 1].isArmed();
-    }
-
-    /**
-     * Set whether or not the zone sensor is armed
-     *
-     * @param zone the index of the zone (first zone is index 1)
-     * @param armed true if armed
-     */
-    public void setSensorArmed(int zone, Boolean armed) {
-        if ((zone >= 1) && (zone <= zones.length)) {
-            this.zones[zone - 1].setArmed(armed);
+    public PowermaxZoneState getZone(int zone) {
+        if ((zone < 1) || (zone > zones.length)) {
+            return new PowermaxZoneState(timeZoneProvider);
```

Comment:
Shouldn't you throw an IllegalArgumentException instead? Returning an untracked PowermaxZoneState instance only serves to hide a bug in your code.

Reply:
Actually, IndexOutOfBoundsException would be a better thing to throw.

Reply:
Thanks for the review. This pattern is used throughout the code (mostly existing): if the serial data refers to an unrecognized or invalid object, just ignore the error and proceed if possible. This protocol is entirely reverse-engineered so if there's something out of bounds, then it's probably the result of unexpected serial data or an incomplete understanding of the protocol, rather than an error in the code. Throwing an exception would just break the binding unnecessarily. The original version silently ignores an invalid array index and this does the same thing, in a slightly different way.

Reply:
You could potentially set a dummy state due to an invalid zone index but then this dummy state will not be used later to set the openHAB thing channels. So I think there is no problem with that, even if it is different to my approach which was to ignore an invalid zone index.

Reply:
Maybe it would be better to at least add a log entry when the zone index value is out of bounds. Can you please add one (WARN level) ?

Reply:
Agree, will do.